core/dtn7/src/routing/erouting/processing.rs

```diff
@@ -0,0 +1,255 @@
+pub fn notify(notification: RoutingNotifcation) {
+    let packet: Packet = match notification {
+        RoutingNotifcation::SendingFailed(bid, cla_sender) => {
+            Packet::SendingFailed(SendingFailed { bid, cla_sender })
+        }
+        RoutingNotifcation::IncomingBundle(bndl) => {
+            Packet::IncomingBundle(IncomingBundle { bndl: bndl.clone() })
+        }
+        RoutingNotifcation::IncomingBundleWithoutPreviousNode(bid, node_name) => {
+            Packet::IncomingBundleWithoutPreviousNode(IncomingBundleWithoutPreviousNode {
+                bid,
+                node_name,
+            })
+        }
+        RoutingNotifcation::EncounteredPeer(eid) => Packet::EncounteredPeer(EncounteredPeer {
+            name: eid.node().unwrap(),
+            eid: eid.clone(),
+            peer: peers_get_for_node(&eid).unwrap(),
+        }),
+        RoutingNotifcation::DroppedPeer(eid) => Packet::DroppedPeer(DroppedPeer {
+            name: eid.node().unwrap(),
+            eid: eid.clone(),
+        }),
+    };
+
+    send_packet(&packet);
+}
+
+fn remove_response_channel(id: &str) {
+    RESPONSES.lock().unwrap().remove(id);
+}
+
+fn create_response_channel(id: &str, tx: oneshot::Sender<Packet>) {
+    RESPONSES.lock().unwrap().insert(id.to_string(), tx);
+}
+
+/// Tries to send a routing requests to the external router and waits for the response.
+/// The wait will be limited to a timeout of 250ms.
+pub async fn sender_for_bundle(bp: &BundlePack) -> (Vec<ClaSenderTask>, bool) {
+    trace!("external sender_for_bundle initiated: {}", bp);
+
+    if CONNECTION.lock().unwrap().is_none() {
+        return (vec![], false);
+    }
+
+    // Register a response channel for the request
+    let (tx, rx) = oneshot::channel();
+    create_response_channel(bp.to_string().as_str(), tx);
+
+    // Send out the SenderForBundle packet
+    let packet: Packet = Packet::SenderForBundle(SenderForBundle {
+        clas: cla_names(),
+        bp: bp.clone(),
+    });
+    send_packet(&packet);
+
+    let res = timeout(time::Duration::from_millis(250), rx).await;
+    if let Ok(Ok(Packet::SenderForBundleResponse(packet))) = res {
+        remove_response_channel(bp.to_string().as_str());
+
+        if packet.bp.to_string() != bp.to_string() {
+            error!("got a wrong bundle pack! {} != {}", bp, packet.bp);
+            return (vec![], false);
+        }
+
+        return (
+            packet
+                .clone()
+                .clas
+                .iter()
+                .filter_map(|sender| {
+                    for cla_instance in &(*CLAS.lock()) {
+                        if sender.agent == cla_instance.name() {
+                            let dest = format!(
+                                "{}:{}",
+                                sender.remote,
+                                sender.port.unwrap_or_else(|| cla_instance.port())
+                            );
+
+                            return Some(ClaSenderTask {
```

Comment:
this is a return within a return, this code seems quite complex and might be expressed differently better

Reply:
Was refactored to other function to make this function less complex.